src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -128,7 +126,15 @@ private[actions] abstract class RefreshActionBase(
             _) =>
           location
             .allFiles()
-            .map(f => FileInfo(f.getPath.toString, f.getLen, f.getModificationTime))
+            .map { f => // for each file, if it already has a file id,
+                        // add that id to its corresponding FileInfo.
+              val filePath = f.getPath.toString
+              previousIndexLogEntry.fileNameToIdMap.get(filePath) match {
```

Comment:
IMO you need to check if the name/size/modification time is in previous log entry or no to handle modified files. Otherwise modified files get old ids, not a new one. This might cause some confusion.

Reply:
I added some clarification to the comment.

Reply:
Can we have existing tests for modified files?

Reply:
@imback82 
> Do we have existing tests for modified files?
We have below test case:
https://github.com/microsoft/hyperspace/blob/master/src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala#L195